sdks/python/apache_beam/runners/portability/fn_api_runner/worker_handlers.py

```diff
@@ -750,48 +745,47 @@ def host_from_worker(self):
 
   def start_worker(self):
     # type: () -> None
-    with SUBPROCESS_LOCK:
-      try:
-        _LOGGER.info('Attempting to pull image %s', self._container_image)
-        subprocess.check_call(['docker', 'pull', self._container_image])
-      except Exception:
-        _LOGGER.info(
-            'Unable to pull image %s, defaulting to local image if it exists' %
-            self._container_image)
-      self._container_id = subprocess.check_output([
-          'docker',
-          'run',
-          '-d',
-          # TODO:  credentials
-          '--network=host',
-          self._container_image,
-          '--id=%s' % self.worker_id,
-          '--logging_endpoint=%s' % self.logging_api_service_descriptor().url,
-          '--control_endpoint=%s' % self.control_address,
-          '--artifact_endpoint=%s' % self.control_address,
-          '--provision_endpoint=%s' % self.control_address,
+    try:
+      _LOGGER.info('Attempting to pull image %s', self._container_image)
+      subprocess.check_call(['docker', 'pull', self._container_image])
```

Comment:
I think it makes sense to keep the lock to avoid concurrent pulls of the same image.
In general, it's better to separate such potentially non-trivial changes in separate PRs, but thanks for putting it in it's own commit.

Reply:
Sorry, I overlooked. Reverted changes concerning this.